src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -0,0 +1,104 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+// TODO: Get rid of this once I figure out how to handle build errors like these given the current lazy creation pattern:
+//     System\Net\Http\HttpTelemetry.cs(19,32): error CS8618: Non-nullable field '_startedRequestsCounter' is uninitialized.
+//     Consider declaring the field as nullable. [S:\GitHub\runtime\src\libraries\System.Net.Http\src\System.Net.Http.csproj]
+#nullable disable
+
+namespace System.Net.Http
+{
+    [EventSource(Name = "System.Net.Http")]
+    internal sealed class HttpTelemetry : EventSource
+    {
+        public static readonly HttpTelemetry Log = new HttpTelemetry();
+
+        private IncrementingPollingCounter _requestsPerSecondCounter;
+        private PollingCounter _startedRequestsCounter;
+        private PollingCounter _currentRequestsCounter;
+        private PollingCounter _abortedRequestsCounter;
+
+        private long _startedRequests;
+        private long _currentRequests;
+        private long _abortedRequests;
+
+        public static new bool IsEnabled => Log.IsEnabled();
+
+        // NOTE
+        // - The 'Start' and 'Stop' suffixes on the following event names have special meaning in EventSource. They
+        //   enable creating 'activities'.
+        //   For more information, take a look at the following blog post:
+        //   https://blogs.msdn.microsoft.com/vancem/2015/09/14/exploring-eventsource-activity-correlation-and-causation-features/
+        // - A stop event's event id must be next one after its start event.
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void RequestStart(string host, int port)
+        {
+            Interlocked.Increment(ref _startedRequests);
+            Interlocked.Increment(ref _currentRequests);
+            WriteEvent(1, host, port);
+        }
+
+        [Event(2, Level = EventLevel.Informational)]
+        public void RequestStop(string host, int port)
+        {
+            Interlocked.Decrement(ref _currentRequests);
+            WriteEvent(2, host, port);
+        }
+
+        [NonEvent]
+        public void RequestAbort()
+        {
+            Interlocked.Increment(ref _abortedRequests);
+        }
+
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                // This is the convention for initializing counters in the RuntimeEventSource (lazily on the first enable command).
+                // They aren't disabled afterwards...
+
+                // The cumulative number of HTTP requests started since the process started.
+                _startedRequestsCounter ??= new PollingCounter("requests-started", this, () => _startedRequests)
+                {
+                    DisplayName = "Requests Started",
```

Comment:
I think the display names should be in resources?

Reply:
I followed this pattern from ASP.NET. They don't use resources for these strings. It is similar perhaps to other logging where we do not localize at all.

@davidfowl @Tratcher Can you comment?